edgedb-protocol/src/query_arg.rs

```diff
@@ -237,24 +220,37 @@ impl QueryArg for Value {
             (LocalDatetime(_), BaseScalar(d)) if d.id == codec::CAL_LOCAL_DATETIME => Ok(()),
             (LocalTime(_), BaseScalar(d)) if d.id == codec::CAL_LOCAL_TIME => Ok(()),
             (RelativeDuration(_), BaseScalar(d)) if d.id == codec::CAL_RELATIVE_DURATION => Ok(()),
-            (Str(_), BaseScalar(d)) if d.id == codec::STD_STR => Ok(()),            
+            (Str(_), BaseScalar(d)) if d.id == codec::STD_STR => Ok(()),
             (Uuid(_), BaseScalar(d)) if d.id == codec::STD_UUID => Ok(()),
+            (Enum(val), Enumeration(EnumerationTypeDescriptor { members, .. })) => {
+                if members.iter().any(|c| c == val.deref()) {
+                    Ok(())
+                } else {
+                    let members = members
+                        .into_iter()
+                        .map(|c| format!("'{c}'"))
+                        .collect::<Vec<_>>()
+                        .join(", ");
+                    let val = val.to_string();
```

Comment:
I think both `val.deref()` and `val.to_string()` can be replaced by a local variable `let val = val.deref()`.

Reply:
True, looks much nicer!